homeassistant/components/mqtt/light/schema_template.py

```diff
@@ -178,9 +176,25 @@ def _setup_from_config(self, config):
             and self._templates[CONF_BLUE_TEMPLATE] is not None
         ):
             color_modes.add(ColorMode.HS)
-        self._supported_color_modes = filter_supported_color_modes(color_modes)
-        if len(self._supported_color_modes) == 1:
-            self._fixed_color_mode = next(iter(self._supported_color_modes))
+        self._attr_supported_color_modes = filter_supported_color_modes(color_modes)
+        self._fixed_color_mode = None
+        if len(self.supported_color_modes) == 1:
+            self._fixed_color_mode = next(iter(self.supported_color_modes))
+
+        features = LightEntityFeature.FLASH | LightEntityFeature.TRANSITION
+        if config.get(CONF_EFFECT_LIST) is not None:
+            features = features | LightEntityFeature.EFFECT
+
+        self._attr_supported_features = features
+        if self._fixed_color_mode:
+            self._attr_color_mode = self._fixed_color_mode
```

Comment:
I think you could adjust the grouping here slightly, by setting `self._attr_color_mode` earlier.
```suggestion
        if len(self.supported_color_modes) == 1:
            self._fixed_color_mode = next(iter(self.supported_color_modes))
            self._attr_color_mode = self._fixed_color_mode

        features = LightEntityFeature.FLASH | LightEntityFeature.TRANSITION
        if config.get(CONF_EFFECT_LIST) is not None:
            features = features | LightEntityFeature.EFFECT
        self._attr_supported_features = features
```